packages/profile-sync-controller/src/controllers/user-storage/UserStorageController.ts

```diff
@@ -205,9 +338,15 @@ export default class UserStorageController extends BaseController<
       );
       this.#isUnlocked = isUnlocked;
 
-      this.messagingSystem.subscribe('KeyringController:unlock', () => {
-        this.#isUnlocked = true;
-      });
+      this.messagingSystem.subscribe(
+        'KeyringController:unlock',
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        async () => {
+          this.#isUnlocked = true;
+
+          await this.syncInternalAccountsWithUserStorage();
```

Comment:
Love this pattern!
GL if we can find an event for extension/mobile open.
I'm still not against a CRON (every 5mins/1hr for eventual consistency).